src/main/java/net/openhft/chronicle/queue/reader/ReadResult.java

```diff
@@ -0,0 +1,21 @@
+package net.openhft.chronicle.queue.reader;
+
+public class ReadResult {
+    public static final ReadResult NOT_FOUND = new ReadResult(false, 0);
+
+    private final boolean found;
+    private final int matches;
+
+    public ReadResult(boolean found, int matches) {
+        this.found = found;
+        this.matches = matches;
+    }
+
+    public boolean found() {
+        return found;
+    }
+
+    public int matches() {
+        return matches;
+    }
```

Comment:
If we make this class mutable, we could hold TL instances to prevent creating new objects. That has other drawbacks of course.

Reply:
Got rid of it